eox_core/api/v1/tests/integration/test_views.py

```diff
@@ -1543,3 +1562,148 @@ def test_info_view_success(self) -> None:
+            "mode": self.mode,
+            "is_active": True,
+            "force": True,
+        }
+        self.params = {
+            "username": self.user['username'],
+            "course_id": self.demo_course_id,
+        }
+
+    @ddt.data(
+        {"detailed": False, "grading_policy": False},
+        {"detailed": True, "grading_policy": False},
+        {"detailed": True, "grading_policy": True},
+    )
+    def test_read_valid_params_successfully(self, extra_params: dict):
+        """
+        Get grades info from a user enrolled on a course.
+
+        Open edX definitions tested:
+        - `get_edxapp_user`
+        - `get_enrollment`
+        - `get_course_grade_factory`
+        - `get_valid_course_key`
+        - `get_courseware_courses`
+        - `grade_factory`
+
+        Expected result:
+        - The status code is 200.
+        - The response includes the 'earned_grade'.
+        - The response includes the 'section_breakdown' if 'detailed' params is True, otherwise, it is not included.
+        - The response includes the 'grading_policy' if 'grading_policy' params is True, otherwise, it is not included.
+        """
+        self.create_enrollment(self.tenant_x, self.create_enrollment_data)
+        params = self.params.copy()
+        params.update(extra_params)
+
+        response = self.get_grade_info(self.tenant_x, params)
+        response_content = response.json()
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertIn("earned_grade", response_content)
+        if extra_params["detailed"]:
+            self.assertIn("section_breakdown", response_content)
+        else:
+            self.assertNotIn("section_breakdown", response_content)
+
+        if extra_params["grading_policy"]:
+            self.assertIn("grading_policy", response_content)
+            self.assertIn("grader", response_content["grading_policy"])
+            self.assertIn("grade_cutoffs", response_content["grading_policy"])
+        else:
+            self.assertNotIn("grading_policy", response_content)
+
+    def test_invalid_user_grade_for_another_site(self):
+        """
+        Test get grade info for a user and course from another site.
+
+        Open edX definitions tested:
+        - `get_edxapp_user`
+
+        Expected result:
+        - The status code is 404.
+        - The response contains an error message.
+        """
+        params = self.params.copy()
+        params.update({
+            "detailed": True,
+            "grading_policy": True,
+        })
+
+        response = self.get_grade_info(self.tenant_y, params)
+        response_data = response.json()
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+        self.assertIn("detail", response_data)
+        self.assertEqual(
+            response_data["detail"],
+            f"No user found by {{'username': '{params['username']}'}} on site {self.tenant_y['domain']}.",
+        )
```

Comment:
We've been using the [AAA (Arrange-Act-Assert) format](https://www.pythonhello.com/practice/python-unit-tests-patterns) for testing, can we do that here and in the other tests as well?

```suggestion
        params = self.params.copy()
        params.update({
            "detailed": True,
            "grading_policy": True,
        })

        response = self.get_grade_info(self.tenant_y, params)
        response_data = response.json()
        
        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
        self.assertIn("detail", response_data)
        self.assertEqual(
            response_data["detail"],
            f"No user found by {{'username': '{params['username']}'}} on site {self.tenant_y['domain']}.",
        )
```